src/CakeSlayer/SqlArray.hs

```diff
@@ -0,0 +1,39 @@
+{- | Introduces a useful wrapper around lists for easier work with DB
+and frontend.
+-}
+
+module CakeSlayer.SqlArray
+       ( SqlArray (..)
+       ) where
+
+import Data.Aeson (FromJSON, ToJSON)
+import Data.Swagger.Schema (ToSchema (..))
+import Database.PostgreSQL.Simple.FromField (FieldParser, FromField (..))
+import Database.PostgreSQL.Simple.ToField (Action, ToField (..))
+import Database.PostgreSQL.Simple.Types (PGArray (..))
+import Elm (Elm (..))
+
+
+{- | Our own wrapper around list of elements. This data type is required,
+because @postgresql-simple@ library doesn't have 'FromField' instance for list.
+But 'PGArray' type doesn't have required instances (json, elm). So we introduce
+our own type, transparent for frontend.
+-}
+newtype SqlArray a = SqlArray
+    { unSqlArray :: [a]
+    } deriving stock (Generic, Show)
+      deriving newtype (Eq, FromJSON, ToJSON)
+
+instance Elm a => Elm (SqlArray a) where
+    toElmDefinition _ = toElmDefinition (Proxy @[a])
```

Comment:
I remember this instance was needed because we hav `OVERLAPPING` instances in `elm-street` but they were removed in last release. Could you try to derive this instance using `newtype` strategy?

Reply:
Oh, sure, sorry!

Reply:
@vrom911 No worries! I should've been left comment near this instance...